modules/phase_field/include/postprocessors/GrainTrackerMerge.h

```diff
@@ -0,0 +1,13 @@
+// Added function: When misorientation is below a certain threshold, grains merge
+
+#pragma once
+
+#include "GrainTracker.h"
+
+class GrainTrackerMerge : public GrainTracker
```

Comment:
I spoke with one of our phase field experts and we believe that all of your new logic should just go in this class to leave the existing classes alone to the extent that's possible. You will have to add your new adjacency members to the `FeatureData` structure, and you may also need a new callback, but most of the rest of this can go in this class to separate it out. 

This will also allow you to have more of a "development" object where you can put your work without affecting the base classes. It'll be easier for your to make changes and get them merged.

Reply:
Yeah, I also agree with that the new logic of determining whether the grains are merged should be included in GrainTrackerMerge.

> You will have to add your new adjacency members to the FeatureData structure, and you may also need a new callback,

I am not quit sure what yuo mean by "a new callback". As you can see, I created the `_adjacent_id` direcly in the `FeatureData` of the `FeatureFloodCount` class. This has not been moved to the new GrainTrackMerge , which I think it would be reasonable now for the following reasons:

1. Once materials or vectorpostprocessors objects may also need to access this part  of the data, such as a vectorpostprocessors class that can obtain the number of adjacent grains of each grain to verify the class  von Neumann-Mullins theory in grain growth.
2. If this part is moved to GrainTrackerMerge, the specific operation is a bit challenging for me.